### PR TITLE
safe-room.lic: Allow customizing which town to use an autopuff in.

### PR DESCRIPTION
**Pull request by Sarvatt**
Fixes: #2451

### CODE REVIEW

File: safe-room.lic [hunk 1]
```diff
@@ -40,7 +40,13 @@ class SafeRoom
       Flags.add('idle', 'you have been idle too long')
 
       town_data = get_data('town')
-      hometown = town_data[settings.hometown]
+      if settings.force_healer_town
+        temp_settings = settings
+        temp_settings.hometown = settings.force_healer_town
```
Comment (rpherbig) [inline safe-room.lic]: Did you consider `hometown = town_data[settings.force_healer_town]`?

File: validate.lic [hunk 2]
```diff
@@ -503,6 +503,18 @@ class DRYamlValidator
...
+  def assert_that_force_healer_town_is_valid(settings)
+    return unless settings.force_healer_town
+
+    valid_hometowns = get_data('town')
+                      .to_h
+                      .keys
+                      .map(&:to_s)
+
+    return if valid_hometowns.include? settings.force_healer_town
+    warn("Your force_healer_town setting of '#{settings.force_healer_town}' is not valid. Valid values are: #{valid_hometowns}")
```
Comment (rpherbig) [inline validate.lic]: Did you consider checking that the hometown also has an empath set?

Comment (Sarvatt) [reply validate.lic]: Done, and tested that it works.

Comment (rpherbig) [reply validate.lic]: Should this be `error` instead of `warn`?

File: validate.lic [hunk 3]
```diff
@@ -503,6 +503,30 @@ class DRYamlValidator
...
+  end
+
+  def assert_that_force_healer_town_has_npc_empath(settings)
+    return unless settings.force_healer_town
+
+    town_data = get_data('town')
+    hometown = town_data[settings.force_healer_town]
+    if hometown['npc_empath'].nil?
+      echo('No npc empath exists in your force_healer_town so healing will not work, please pick another town.')
```
Comment (rpherbig) [inline validate.lic]: Should this be `error` instead of `echo`?

File: validate.lic [hunk 4]
```diff
@@ -503,6 +503,30 @@ class DRYamlValidator
...
+                      .to_h
+                      .keys
+                      .map(&:to_s)
+
+    return if valid_hometowns.include? settings.force_healer_town
+    warn("Your force_healer_town setting of '#{settings.force_healer_town}' is not valid. Valid values are: #{valid_hometowns}")
+  end
+
+  def assert_that_force_healer_town_has_npc_empath(settings)
+    return unless settings.force_healer_town
```
Comment (rpherbig) [inline validate.lic]: Should this test be generalized to also check `settings.hometown` if there is no `force_healer_town`?

**Comment by Sarvatt** [comment]
Echo changed to error, warn changed to error, and I made the npc empath validation check generic so it works for all settings. Verified it works by trying both hometown: Arthe Dale and force_healer_town: Arthe Dale independently.

**Pull request merged**